tasks/image-augmentation/src/utils.py

```diff
@@ -43,3 +51,22 @@ def copySample(sample: ImageSample, dataset: ImageDataset) -> None:
 
             if not copy.delete():
                 logging.error("\tFailed to delete sample")
+
+
+def getOutputDatasetName(taskRun: TaskRun) -> str:
+    relevantParams = taskRun.parameters.copy()
+
+    relevantParams["dataset"] = relevantParams["dataset"].id
+    relevantParams.pop("outputDataset")
+
+    return hashCacheName(f"{taskRun.id}-AugImg", ".".join(str(relevantParams.values())))
+
+
+def getCache(cacheName: str, expectedSize: int) -> Optional[ImageDataset]:
+    caches = ImageDataset.fetchAll(name = cacheName, include_sessions = 1)
```

Comment:
Let's use cache specific functions to create and fetch cache like `fetchCachedDataset` and `createCacheDataset`